src/EventListener/LockMigrationsListener.php

```diff
@@ -59,7 +65,44 @@ private function shouldSkip(ConsoleEvent $event): bool
 
     private function getConnectionName(ConsoleEvent $event): string
     {
-        return (string)$event->getInput()->getOption('conn');
+        $connectionName = (string) $event->getInput()->getOption('conn');
+
+        if ('' === $connectionName) {
+            try {
+                $entityManager = $this->dependencyFactory->getEntityManager();
+            } catch (MissingDependency $e) {
+                if ($e->getMessage() !== MissingDependency::noEntityManager()->getMessage()) {
+                    throw $e;
+                }
+
+                $entityManager = null;
+            }
+
+            $connectionName = $entityManager
+                ? $this->getEntityManagerConnectionName($entityManager)
+                : null;
+
+            return $connectionName
+                ?? $this->dependencyFactory->getConfiguration()->getConnectionName()
```

Comment:
Did you test this in your project? During `onConsoleCommand` this should not be available afaik, because the config is only loaded from within the doctrine migrations command.